### PR TITLE
feat: Added post-deployment test support

### PR DESCRIPTION
**Pull request by AlexanderSehr**
## Description

> **Note:** Test file changes for key vault are just for illustration purposes

- Introduced 2 additional steps to deploy action that will look for Pester test files anywhere in a e2e test's folder (or nested folder, if desired), run the contained tests and publish the results
- If no test was found, the publish results step will be skipped
- By default, the action will pass in one parameter to the test files that MUST be implemented: `[hashtable] $TestInputData`. This parameter will automatically contain the test file outputs + the path to the test file. For example
  ```json
  {
    "TestInputData": {
      "ModuleTestFolderPath": "/home/runner/work/bicep-registry-modules/bicep-registry-modules/avm/res/key-vault/vault/tests/e2e/private-endpoint",
      "DeploymentOutputs": {
        "resourceId": {
          "type": "string",
          "value": "/subscriptions/1111111/resourcegroups/(...)"
        } 
      }
    }
  }
  ```
- Example [run](https://github.com/AlexanderSehr/bicep-registry-modules/actions/runs/7944560531/job/21690223560)

| Pipeline |
| - |
| [avm.res.key-vault.vault](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.key-vault.vault.yml) |

### CODE REVIEW

File: avm/res/key-vault/vault/tests/e2e/private-endpoint/main.test.bicep
Comment (AlexanderSehr) [inline avm/res/key-vault/vault/tests/e2e/private-endpoint/main.test.bicep:103]: I'd suggest to remove these tests. But should we maybe add some for a module owned by a maintainer? @ChrisSidebotham / @eriqua?

Comment (eriqua) [reply avm/res/key-vault/vault/tests/e2e/private-endpoint/main.test.bicep:103]: You mean remove all added pe tests? Agree, and sure for adding them to a module owned by a maintainer. Easy for me to say as I don't own one :smile:

Comment (eriqua) [reply avm/res/key-vault/vault/tests/e2e/private-endpoint/main.test.bicep:103]: I anyway guess the real implementation example will come with the first pattern module

**Review by fblix — DISMISSED** [review]

**Review by ChrisSidebotham — DISMISSED** [review]
lgtm

File: avm/res/key-vault/vault/tests/e2e/private-endpoint/main.test.bicep
Comment (AlexanderSehr) [reply avm/res/key-vault/vault/tests/e2e/private-endpoint/main.test.bicep:103]: Got sign-off by @fblix . Beautified & re-running tests

**Review by ChrisSidebotham — APPROVED** [review]
🤌

**Pull request merged**
